### PR TITLE
refactor(router): remove deprecated feature and related old code

### LINKED ISSUE #14648: envoy.reloadable_features.preserve_query_string_in_path_redirects deprecation
Your change #11493 (feat(router): keep query string by default for redirects) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.
cc @knight42

### PR DESCRIPTION
**Pull request by knight42**
Signed-off-by: knight42 <anonymousknight96@gmail.com>

Commit Message: refactor(router): remove dprecated runtime feature and related old code.
Additional Description: It has been 6 monthes since `envoy.reloadable_features.preserve_query_string_in_path_redirects` was introduced, and this runtime feature is deprecated now.
Risk Level: N/A
Testing: N/A
Docs Changes: N/A
Release Notes: Runtime feature `envoy.reloadable_features.preserve_query_string_in_path_redirects` is removed now.
Platform Specific Features: N/A
[Optional Runtime guard:]
Fixes #14648
[Optional Deprecated:]

### CODE REVIEW

**Review by asraa — COMMENTED** [review]
Just a drive-by: Could you please add a release note (`/docs/root/version_history/current.rst`) about the removed feature?

(You may also want to remove the feature from `runtime_features.cc` and from `config_impl_speed_test.cc`)

File: source/common/router/config_impl.cc [hunk 1]
```diff
@@ -332,8 +332,6 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
                          : ""),
       path_redirect_(route.redirect().path_redirect()),
       path_redirect_has_query_(path_redirect_.find('?') != absl::string_view::npos),
-      enable_preserve_query_in_path_redirects_(Runtime::runtimeFeatureEnabled(
```
Comment (asraa) [inline source/common/router/config_impl.cc:335]: One last comment -- I think you can remove `#include "common/runtime/runtime_features.h"`

**Review by asraa — CHANGES_REQUESTED** [review]
Thanks! Could you please avoid force pushing? It breaks reviewer flow.

File: source/common/router/config_impl.cc (hunk 1 above)
Comment (knight42) [reply source/common/router/config_impl.cc:335]: oops sorry about force pushing... I have removed this line.

**Review by asraa — APPROVED** [review]
Thanks! LGTM

**Comment by mattklein123** [comment]
I'm confused but I think the coverage issue is legitimate. Can you take a look? You can look at the coverage report through CI and also compare it to the main branch coverage: https://storage.googleapis.com/envoy-postsubmit/master/coverage/index.html

/wait

**Comment by knight42** [comment]
I am confused as well.. The presubmit job failed because: ([link](https://dev.azure.com/cncf/envoy/_build/results?buildId=63455&view=logs&j=5035ce29-be3c-5fcf-dc50-abb6a265f8b1&t=548c3e59-0417-5971-4f1b-f8545005c291&l=17515))
```
Per-extension coverage failed:
Code coverage for source/extensions/transport_sockets/tap is lower than limit of 96.6 (95.7)
```
I cannot think of the reason why the coverage of `source/extensions/transport_sockets/tap` should be affected by this PR. I will take a closer look later.

**Pull request closed without merging**